### PR TITLE
Revert broadcasting `lint_result_changed` cache layer

### PR DESCRIPTION
**Pull request by kaste**
Generally, we can only apply such broad cache layers if the render
functions are "pure", i.e. only draw the highlights if we actually
have different errors from before, otherwise they're already in place.

This breaks for the highlights often because of #1617 or rather the
underlying https://github.com/SublimeTextIssues/Core/issues/2877

I.e. whenever Sublime invalidates all drawn regions (usually because a
file gets reloaded), we can retrigger a lint but still get the same
results as before. In that case we *need* to draw because all regions
are actual dangles but the cache would prevent this. Basically our
error store and the view state are out-of-sync. This is solvable
if we can solve #1617.

For the panel, the cache is just not applicable at all. The intention
was to solve #1623 "for free" in a pragmatic way. But that's just not
the case, and actually we're breaking the `show_panel_on_save` feature.

The usage in `status_bar_view` was never important performance wise
so we can, for now, just revert the whole thing.

Partially reverts 816904c87f8eae497872fc422b07dc43781ff7af
Reopens #1623

### CODE REVIEW

**Comment by braver** [comment]
So would this bring back dangling highlights, or does it simply mean we don't have a really good solution? I mean, usually not having a cache is a mostly a performance hit, but do we regress functionally here? I don't see any immediate downsides here, but the dangling stuff is quite subtle.

**Comment by kaste** [comment]
We regress with the cache.  

As soon as we get dangles, we *always* need to redraw. Sure most of the time, we have different results from the linter, Sublime reloads the file because you reverted changes or whatever so there is a good *chance* that it has different errors, and then it will draw of course because new errors can't match the cache. But if we have the same results after dangling (you revert something at the end of the file but had import errors on the top)... it is possible that we have dangles *until* the user edits enough so that the linter reports *different* errors. It is not super likely, it surely doesn't happen *all the time* but it can. 

Because it is rather subtle a user will likely roll eyes, feel insecure about the software. You know, it's not right but it heals itself after editing.  Unlikely we will get reports about such things because these are common in this software world. 

tldr; The view state and the store are out of sync after the dangles. T.i. we're not pure 
(in the simple sense of the view reflects the data) and if we're not pure we can't cache. So imo we have to revert (until we solve #1617).  It's that simple: if only *we* write the regions we're safe of course, but if Sublime modifies the view as well and we don't detect it 🐛

**Comment by braver** [comment]
> Because it is rather subtle a user will likely roll eyes, feel insecure about the software. You know, it's not right but it heals itself after editing.

Totally agree here. My point was more that it's hard to test for if you don't know what to look for. The original dangles and zombies we only found after scratching our heads for a few weeks trying to understand what we were seeing. So I find it hard to test what the effect is of this PR. 

The general idea that we cannot cache here is totally clear, so I'm all on board with the intent of the PR. So the question is are you confident here (the changeset is small enough we can basically read it and know that it reverts the right things), or if you want me to also test this for a few days to try and find quirks?

**Comment by kaste** [comment]
Actually, I did not reproduce the bug. I just thought: shit I forgot the dangles! 😁  (I think our store actually knows where the regions/dangles are because in the gif at #1617 it shows that SublimeLinter updates the panel and shows all errors on 1:1 position, which is already super magic and awesome, but our cache doesn't produce a new hash after that.) I just know that I wasn't full-capacity while doing the commit in the first place which also counts.

There is nothing to test here. If there is no typo, we just subscribe to the old event we always used.

**Comment by braver** [comment]
Ok then. Let's just roll with it then.
